master/internal/project/postgres_project.go

```diff
@@ -344,6 +344,20 @@ func UpdateProject(
 			return nil
 		}
 
+		var isUsed bool
+		err = tx.NewSelect().Table("local_id_redirect").
+			ColumnExpr("COUNT(*) > 0 as is_used").
+			Where("project_key = ?", currentProject.Key).
```

Comment:
Does this mean project key is one time use? For example, if a user change project key from "AAA" to "BBB", then it's not possible to use "AAA" as key anymore, even though "AAA" is not currently associated with any project?

Reply:
Yes, the redirect table will need to allow us to make old links work, so once a key is used it cannot be used again.

Reply:
How does this get handled if a project starts with the key "AAA" -> "BBB" and then an attempt is made to change it back to "AAA"? Does it make sense to not allow a project to reuse one of it's own keys?

Reply:
Updated so that we check for the project id. If the project id of the matching key is the same we allow it to be changed back.

Reply:
Would it be worth limiting the number of keys that can be associated with a project at a given point in time? Like "we'll only save the history of the last 5 project keys"? 

This can be a discussion that's out of scope for this PR, and might not even be necessary if we're not seeing performance concerns.